### PR TITLE
Logging Enhancement

### PR DESCRIPTION
**Pull request by openbots-ff**
- Responsibility of Logger instantiation has been moved out of the AutomationEngineInstance and into Studio or wherever necessary.
- User can now set the Engine Logger in Studio Settings to File, HTTP, or SignalR and set their respective parameters. 
- User can explicitly write logs using the Log Message Command.
- Necessary modifications to the Documentation Generation have been added to account for the changes made in the command cleanup.

### CODE REVIEW

File: taskt.Studio/Commands/Engine Commands/LogMessageCommand.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        [PropertyUISelectionOption("Verbose")]
+        [PropertyUISelectionOption("Debug")]
+        [PropertyUISelectionOption("Information")]
+        [PropertyUISelectionOption("Warning")]
+        [PropertyUISelectionOption("Error")]
+        [PropertyUISelectionOption("Fatal")]
+        [InputSpecification("Specify the log type.")]
+        [SampleUsage("")]
+        [Remarks("")]
+        [PropertyUIHelper(UIAdditionalHelperType.ShowVariableHelper)]
```
Comment (Ali-Basharat) [inline taskt.Studio/Commands/Engine Commands/LogMessageCommand.cs:53]: I think we do not provide a VariableHelper for a dropdown.

File: taskt.Studio/Commands/Engine Commands/LogMessageCommand.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        public string v_LogType { get; set; }
```
Comment (ob-ab) [inline taskt.Studio/Commands/Engine Commands/LogMessageCommand.cs]:
```suggestion
        [Remarks("")]
        public string v_LogType { get; set; }
```

**Review by ob-ab — CHANGES_REQUESTED** [review]
Please make a suggested change as I remember that we do not provide a VariableHelper for a Dropdown.

**Review by CameronHerwig — DISMISSED** [review]

**Review by ob-ab — APPROVED** [review]
Please make a suggested change.

**Review by CameronHerwig — APPROVED** [review]

**Pull request merged**